massa-execution/src/types.rs

```diff
@@ -22,6 +25,8 @@ pub(crate) struct StepHistoryItem {
 
     // list of SCE ledger changes caused by this execution step
     pub ledger_changes: SCELedgerChanges,
+
+    pub events: HashMap<Hash, SCOutputEvent>,
```

Comment:
it might be more practical for the user to keep the order of the emitted messages. Maybe ` Vec<(Hash, SCOutputEvent)> `would be nice. What do you think ?

Reply:
This is a place holder for a struct with multiple indexes as we want to search by caller, smart contract and slot. I'd rather add an index field in the SCOutputEvent

Reply:
let's do that